src/pkg/project/models/project.go

```diff
@@ -15,14 +15,192 @@
+	ProjectPrivate = "private"
+)
+
+func init() {
+	orm.RegisterModel(&Project{})
+}
+
+// Project holds the details of a project.
+type Project struct {
+	ProjectID    int64                  `orm:"pk;auto;column(project_id)" json:"project_id"`
+	OwnerID      int                    `orm:"column(owner_id)" json:"owner_id"`
+	Name         string                 `orm:"column(name)" json:"name" sort:"default"`
+	CreationTime time.Time              `orm:"column(creation_time);auto_now_add" json:"creation_time"`
+	UpdateTime   time.Time              `orm:"column(update_time);auto_now" json:"update_time"`
+	Deleted      bool                   `orm:"column(deleted)" json:"deleted"`
+	OwnerName    string                 `orm:"-" json:"owner_name"`
+	Role         int                    `orm:"-" json:"current_user_role_id"`
+	RoleList     []int                  `orm:"-" json:"current_user_role_ids"`
+	RepoCount    int64                  `orm:"-" json:"repo_count"`
+	ChartCount   uint64                 `orm:"-" json:"chart_count"`
+	Metadata     map[string]string      `orm:"-" json:"metadata"`
+	CVEAllowlist allowlist.CVEAllowlist `orm:"-" json:"cve_allowlist"`
+	RegistryID   int64                  `orm:"column(registry_id)" json:"registry_id"`
+}
+
+// GetMetadata ...
+func (p *Project) GetMetadata(key string) (string, bool) {
+	if len(p.Metadata) == 0 {
+		return "", false
+	}
+	value, exist := p.Metadata[key]
+	return value, exist
+}
+
+// SetMetadata ...
+func (p *Project) SetMetadata(key, value string) {
+	if p.Metadata == nil {
+		p.Metadata = map[string]string{}
+	}
+	p.Metadata[key] = value
+}
+
+// IsPublic ...
+func (p *Project) IsPublic() bool {
+	public, exist := p.GetMetadata(ProMetaPublic)
+	if !exist {
+		return false
+	}
+
+	return isTrue(public)
+}
+
+// IsProxy returns true when the project type is proxy cache
+func (p *Project) IsProxy() bool {
+	return p.RegistryID > 0
+}
+
+// ContentTrustEnabled ...
+func (p *Project) ContentTrustEnabled() bool {
+	enabled, exist := p.GetMetadata(ProMetaEnableContentTrust)
+	if !exist {
+		return false
+	}
+	return isTrue(enabled)
+}
+
+// VulPrevented ...
+func (p *Project) VulPrevented() bool {
+	prevent, exist := p.GetMetadata(ProMetaPreventVul)
+	if !exist {
+		return false
+	}
+	return isTrue(prevent)
+}
+
+// ReuseSysCVEAllowlist ...
+func (p *Project) ReuseSysCVEAllowlist() bool {
+	r, ok := p.GetMetadata(ProMetaReuseSysCVEAllowlist)
+	if !ok {
+		return true
+	}
+	return isTrue(r)
+}
+
+// Severity ...
+func (p *Project) Severity() string {
+	severity, exist := p.GetMetadata(ProMetaSeverity)
+	if !exist {
+		return ""
+	}
+	return severity
+}
+
+// AutoScan ...
+func (p *Project) AutoScan() bool {
+	auto, exist := p.GetMetadata(ProMetaAutoScan)
+	if !exist {
+		return false
+	}
+	return isTrue(auto)
+}
+
+// FilterByPublic returns orm.QuerySeter with public filter
+func (p *Project) FilterByPublic(ctx context.Context, qs orm.QuerySeter, key string, value interface{}) orm.QuerySeter {
```

Comment:
Is this being used? It doesn't look correct under the structure we agreed on.

Reply:
cc @heww

Reply:
I didn't know this was based on convention.  
I'm fine with this approach.